common-npm-packages/azure-arm-rest-v2/azure-arm-storage.ts

```diff
@@ -217,6 +218,33 @@ export class StorageAccounts {
         return "";
     }
 
+    private async getStorageAccountsByUri(uri: string, options?: any): Promise<Model.StorageAccount[]> {
```

Comment:
Optimization idea: 
you can also add an optional name parameter here. (e.g. `filterName?: string`)

If `filterName` defined:

You can search parameter in `storageAccounts` in first call. If you find name in first call; you don't need to fetch `nextLink`. If it is not found you can search in nextLink result. 
And return empty array (not found) or 1 item array.

If `filterName` is not defined:
it can work as it is (return all storage accounts in array)

Reply:
Added optimization approach